logsmanagement/logsmanagement.c

```diff
@@ -0,0 +1,1187 @@
+                }
+            }
+
+            /* flb_complete_item_timer_timeout_cb() is needed for 
+             * both local and non-local sources. */
+            p_file_info->flb_tmp_buff_cpy_timer.data = p_file_info;
+            if(unlikely(0 != uv_mutex_init(&p_file_info->flb_tmp_buff_mut))){
+                fatal("uv_mutex_init(&p_file_info->flb_tmp_buff_mut) failed");
+            }
+            uv_timer_init(main_loop, &p_file_info->flb_tmp_buff_cpy_timer);
+            uv_timer_start( &p_file_info->flb_tmp_buff_cpy_timer, 
+                            (uv_timer_cb)flb_complete_item_timer_timeout_cb, 
+                            0, p_file_info->update_timeout * MSEC_PER_SEC);            
+            break;
+        }
+        default: 
+            return p_file_info_destroy(p_file_info);
+    }
+
+
+    /* -------------------------------------------------------------------------
+     * Allocate and initialise parser mutex.
+     * ------------------------------------------------------------------------- */
+    p_file_info->parser_metrics_mut = callocz(1, sizeof(uv_mutex_t));
+    if(unlikely(uv_mutex_init(p_file_info->parser_metrics_mut)))
+        fatal("Failed to initialise parser_metrics_mut for %s", p_file_info->chart_name);
+
+
+    /* -------------------------------------------------------------------------
+     * All set up successfully - add p_file_info to list of all p_file_info structs.
+     * ------------------------------------------------------------------------- */
+    p_file_infos_arr->data = reallocz(p_file_infos_arr->data, (++p_file_infos_arr->count) * (sizeof p_file_info));
+    p_file_infos_arr->data[p_file_infos_arr->count - 1] = p_file_info;
+
+    collector_info("[%s]: initialization completed", p_file_info->chart_name);
+}
+
+static void logsmanagement_main_cleanup(void *ptr) {
+    struct netdata_static_thread *static_thread = (struct netdata_static_thread *)ptr;
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITING;
+
+    collector_info("cleaning up...");
+
+    flb_terminate();
+
+    if(p_file_infos_arr){
+        for(int i = 0; i < p_file_infos_arr->count; i++){
+            p_file_info_destroy(p_file_infos_arr->data[i]);
+        }
+        freez(p_file_infos_arr);
+        p_file_infos_arr = NULL;
+    }
+
+    // TODO: Additional work to do here on exit? Maybe flush buffers to DB?
+
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITED;
+}
+
+/**
+ * @brief The main function of the program.
+ * @details Any static asserts are most likely going to be inluded here. After 
+ * any initialisation routines, the default uv_loop_t is executed indefinitely. 
+ * @todo Any cleanup required on program exit? 
+ */
+void *logsmanagement_main(void *ptr) {
+    netdata_thread_cleanup_push(logsmanagement_main_cleanup, ptr);
+
+    Flb_socket_config_t *forward_in_config = NULL;
+
+    uv_loop_t *main_loop = mallocz(sizeof(uv_loop_t));
+    fatal_assert(uv_loop_init(main_loop) == 0);
```

Comment:
Can this `fatal_assert` to shutdown the whole netdata? If so, please, modify for we shutdown only the thread.

Reply:
After a short talk, I am marking this as resolved.